daemon/test/Makefile

```diff
@@ -0,0 +1,170 @@
+# ----------------------------------------------------------- -*- mode: sh -*-
+#  Directory definitions and pkg-config
+# ----------------------------------------------------------------------------
+
+# Tests specific paths
+_TESTDATA       ?= /tmp/sailjail-daemon-tests
+_TESTSDIR       ?= /opt/tests
+_BINDIR         ?= $(_TESTSDIR)/sailjail-daemon/bin
+_TESTDATADIR    ?= $(_TESTSDIR)/sailjail-daemon/data
+_PREFIX         ?= $(_TESTSDIR)/sailjail-daemon/usr
+_SYSCONFDIR     ?= $(_TESTSDIR)/sailjail-daemon/etc
+_LOCALSTATEDIR  ?= $(_TESTDATA)/var
+
+include ../Makefile.inc
+
+CPPFLAGS += -DTESTDATADIR='"${_TESTDATADIR}"' -I..
+
+# ----------------------------------------------------------------------------
+#  Tests
+# ----------------------------------------------------------------------------
+
+TARGETS_BIN += test_appinfo
+TARGETS_BIN += test_permissions
+TARGETS_BIN += test_settings
+TARGETS_BIN += test_stringset
+TARGETS_BIN += test_util
+
+TARGETS += ${TARGETS_BIN}
+
+# ----------------------------------------------------------------------------
+#  Static test data
+# ----------------------------------------------------------------------------
+
+DATA_FILES += data/keyfile1.ini
+DATA_FILES += data/keyfile2.ini
+DATA_FILES += data/user-1000.settings
+
+APPLICATIONS_FILES += $(wildcard sailjail/applications/*.desktop)
+
+# ----------------------------------------------------------------------------
+# Top level targets
+# ----------------------------------------------------------------------------
+
+.PHONY: build install install-static clean mostlyclean
+
+build:: $(TARGETS)
+
+install:: build install-static install-tests
+	install -d ${DESTDIR}${_BINDIR}
+	install -m755 ${TARGETS_BIN} ${DESTDIR}${_BINDIR}
+
+install-static:: install-data install-applications
+
+clean:: mostlyclean
+	$(RM) $(TARGETS)
+
+mostlyclean::
+	$(RM) *.o *~ *.bak */*.o */*~ */*.bak *.gcda *.gcno
+	$(RM) -r $(PWD)/coverage $(PWD)/report/
+
+# ----------------------------------------------------------------------------
+# Testing targets
+# ----------------------------------------------------------------------------
+
+.PHONY: run coverage coverage-run coverage-build
+
+# Runs without coverage
+run:: _SYSCONFDIR := $(CURDIR)
+run:: _DATADIR := $(CURDIR)/sailjail
+run:: _TESTDATADIR := $(CURDIR)/data
+run:: build copy-test-data run-tests
+
+# Runs with coverage
+coverage-run:: _SYSCONFDIR := $(CURDIR)
+coverage-run:: _DATADIR := $(CURDIR)/sailjail
+coverage-run:: _TESTDATADIR := $(CURDIR)/data
+coverage-run:: coverage-build copy-test-data run-tests
+
+# Builds with coverage on
+coverage-build:: CFLAGS += --coverage
+coverage-build:: build
+
+# Generates report
+coverage:: coverage-run
+	lcov -c -d . -b $(CURDIR) -o $(PWD)/coverage
+	genhtml -o $(PWD)/report $(PWD)/coverage
+	xdg-open $(PWD)/report/index.html
```

Comment:
Isn't this going a bit too far? Usually the path is only reported that can be then opened  by the user. Alternatively, an option for it could be that enables this  and by default it only reports?

Reply:
Ah, this. It was convenient for me but yeah, I should have had echo or something instead. I will change it. :+1: